### PR TITLE
MINOR: Fix config documentation formatting

### PR DESCRIPTION
**Pull request by joel-hamill**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by joel-hamill** [comment]
cc @guozhangwang

File: clients/src/main/java/org/apache/kafka/common/config/internals/BrokerSecurityConfigs.java [hunk 1]
```diff
@@ -61,10 +61,10 @@ public class BrokerSecurityConfigs {
...
             "names to short names (typically operating system usernames). The rules are evaluated in order and the " +
             "first rule that matches a principal name is used to map it to a short name. Any later rules in the list are " +
-            "ignored. By default, principal names of the form {username}/{hostname}@{REALM} are mapped to {username}. " +
-            "For more details on the format please see <a href=\"#security_authz\"> security authorization and acls</a>. " +
-            "Note that this configuration is ignored if an extension of KafkaPrincipalBuilder is provided by the " +
-            "<code>" + PRINCIPAL_BUILDER_CLASS_CONFIG + "</code> configuration.";
+            "ignored. By default, principal names of the form "<code>" + {username}/{hostname}@{REALM} "</code>" + are mapped +
+            "to "<code>" + {username}+ "</code>". For more details on the format please see <a href=\"#security_authz\"> +
+            "security authorization and acls</a>. Note that this configuration is ignored if an extension of +
+            "KafkaPrincipalBuilder is provided by the <code>" + PRINCIPAL_BUILDER_CLASS_CONFIG + "</code> configuration.";
```
Comment (jsancio) [inline clients/src/main/java/org/apache/kafka/common/config/internals/BrokerSecurityConfigs.java]: `KafkaPrincipalBuilder` should be surrounded by `<code></code>`.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 2]
```diff
@@ -481,7 +481,7 @@ public class StreamsConfig extends AbstractConfig {
     @SuppressWarnings("WeakerAccess")
     public static final String CLIENT_ID_CONFIG = CommonClientConfigs.CLIENT_ID_CONFIG;
     private static final String CLIENT_ID_DOC = "An ID prefix string used for the client IDs of internal consumer, producer and restore-consumer," +
-        " with pattern '<client.id>-StreamThread-<threadSequenceNumber>-<consumer|producer|restore-consumer>'.";
+        " with pattern "<code>" + <client.id>-StreamThread-<threadSequenceNumber>-<consumer|producer|restore-consumer> + "</code>.";
```
Comment (jsancio) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: It should be something like:
```java
        " with pattern <code>&lt;client.id&gt;-StreamThread-&lt;threadSequenceNumber$gt;-&lt;consumer|producer|restore-consumer&gt;</code>.";
```

**Review by jsancio — COMMENTED** [review]
Thanks for the changes @joel-hamill

**Comment by joel-hamill** [comment]
@jsancio updated

**Review by jsancio — DISMISSED** [review]
LGTM

File: clients/src/main/java/org/apache/kafka/common/config/internals/BrokerSecurityConfigs.java [hunk 3]
```diff
@@ -61,10 +61,10 @@ public class BrokerSecurityConfigs {
     public static final String SASL_KERBEROS_PRINCIPAL_TO_LOCAL_RULES_DOC = "A list of rules for mapping from principal " +
             "names to short names (typically operating system usernames). The rules are evaluated in order and the " +
             "first rule that matches a principal name is used to map it to a short name. Any later rules in the list are " +
-            "ignored. By default, principal names of the form {username}/{hostname}@{REALM} are mapped to {username}. " +
-            "For more details on the format please see <a href=\"#security_authz\"> security authorization and acls</a>. " +
-            "Note that this configuration is ignored if an extension of KafkaPrincipalBuilder is provided by the " +
-            "<code>" + PRINCIPAL_BUILDER_CLASS_CONFIG + "</code> configuration.";
+            "ignored. By default, principal names of the form <code>{username}/{hostname}@{REALM}</code> are mapped +
+            "to <code>{username}</code>. For more details on the format please see <a href=\"#security_authz\"> +
+            "security authorization and acls</a>. Note that this configuration is ignored if an extension of +
```
Comment (jsancio) [inline clients/src/main/java/org/apache/kafka/common/config/internals/BrokerSecurityConfigs.java]: These 3 lines are missing an end quote. E.g.
```java
            "ignored. By default, principal names of the form <code>{username}/{hostname}@{REALM}</code> are mapped " +
            "to <code>{username}</code>. For more details on the format please see <a href=\"#security_authz\"> " +
            "security authorization and acls</a>. Note that this configuration is ignored if an extension of " +
```

**Review by jsancio — APPROVED** [review]
LGTM

**Pull request merged**